core/src/main/java/hudson/XmlFile.java

```diff
@@ -304,6 +306,9 @@ public Eureka(String encoding) {
         try (InputStream in = Files.newInputStream(file.toPath())) {
             InputSource input = new InputSource(file.toURI().toASCIIString());
             input.setByteStream(in);
+            // this is no real issue, it is just a false positive, since jenkins handles it differently
+            // for more information check: https://github.com/jenkinsci/jenkins/pull/4779
+            JAXP.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
```

Comment:
Also use the feature "http://apache.org/xml/features/disallow-doctype-decl" here. Especially here it shouldn't be a problem.